tests/kafkatest/services/performance/producer_performance.py

```diff
@@ -71,11 +90,15 @@ def parse_stats(line):
                 'latency_99th_ms': float(parts[6].split()[0]),
                 'latency_999th_ms': float(parts[7].split()[0]),
             }
+
         last = None
+
+        self.logger.debug("Producer performance %d command: %s", idx, cmd)
         producer_output = node.account.ssh_capture(cmd)
-        first_line = producer_output.next()
+        wait_until(lambda: self.alive(node), timeout_sec=10, err_msg=self.__class__.__name__ + " took too long to start.")
```

Comment:
Same thing here with not blocking for output, which means you may be starting the JMX tool early and if it fails to connect to the target process, it just exits.